scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -938,6 +949,41 @@ def apply_filters(self, **kwargs):
                 # handle quarantined tests
                 self.handle_quarantined_tests(instance, plat)
 
+                # Check if scripting is available
+                if self.scripting:
+                    matched_scripting = self.scripting.get_matched_scripting(instance.testsuite.id, plat.name)
+                    if matched_scripting:
+                        # Define a function to validate if the platform is supported by the matched scripting
+                        def validate_boards(platform_scope, platform_from_yaml):
+                            return any(board in platform_scope for board in platform_from_yaml)
+
+                    # Define the types of scripts we are interested in as a set
+                    script_types = {'pre_script': 'pre_script_timeout', 'post_flash_script': 'post_flash_timeout', 'post_script': 'post_script_timeout'}
+                    # Iterate over all DUTs to set the appropriate scripts if they match the platform and are supported
+                    for dut in self.env.hwm.duts:
+                        # Check if the platform matches and if the platform is supported by the matched scripting
+                        if dut.platform == plat.name and validate_boards(plat.name, matched_scripting.platforms):
```

Comment:
handle several matched entries case here to check either a single match, or only one `override` element is present which becomes the resulting entry. All other cases will be configuration error.